bin/foreman_scap_client_bash

```diff
@@ -0,0 +1,262 @@
+  # Absence of a variable will cause the script to exit
+  # For empty values, the variable must be set to an empty string
+  POLICY_PROFILE="POLICY_${POLICY_ID}_PROFILE"
+  POLICY_PROFILE=${!POLICY_PROFILE?Policy id $POLICY_ID not found}
+  POLICY_CONTENT_PATH="POLICY_${POLICY_ID}_CONTENT_PATH"
+  POLICY_CONTENT_PATH=${!POLICY_CONTENT_PATH?Policy id $POLICY_ID not found}
+  POLICY_DOWLOAD_PATH="POLICY_${POLICY_ID}_DOWLOAD_PATH"
+  POLICY_DOWLOAD_PATH=${!POLICY_DOWLOAD_PATH?Policy id $POLICY_ID not found}
+  POLICY_TAILORING_PATH="POLICY_${POLICY_ID}_TAILORING_PATH"
+  POLICY_TAILORING_PATH=${!POLICY_TAILORING_PATH?Policy id $POLICY_ID not found}
+  POLICY_TAILORING_DOWLOAD_PATH="POLICY_${POLICY_ID}_TAILORING_DOWLOAD_PATH"
+  POLICY_TAILORING_DOWLOAD_PATH=${!POLICY_TAILORING_DOWLOAD_PATH?Policy id $POLICY_ID not found}
+}
+
+http_proxy_uri () {
+  if [ -n "$HTTP_PROXY_SERVER" ] && [ -n "$HTTP_PROXY_PORT" ]; then
+    echo "http://$HTTP_PROXY_SERVER:$HTTP_PROXY_PORT"
+  else
+    echo ""
+  fi
+}
+
+scan_command_env_vars () {
+  local env_vars=""
+  if [ -n "$(http_proxy_uri)" ]; then
+    env_vars="HTTP_PROXY=$(http_proxy_uri) HTTPS_PROXY=$(http_proxy_uri)"
+  fi
+  echo $env_vars
+}
+
+# Params: version1 version2
+ver_ge () {
+  printf '%s\n%s' "$1" "$2" | sort -C -V -r && echo "true" || echo "false"
+}
+
+supports_local_file_option () {
+  local versions=$(oscap --version)
+  local status=$?
+  if [ $status -ne 0 ]; then
+    echo "false"
+  fi
+  local version=$(echo $versions | awk '{print $NF}' | head -n 1)
+  # OpenSCAP 1.3.6 and newer requires the `--local-files` option to use local copies of remote SDS components
+  if [ "$(ver_ge $version 1.3.6)" == "true" ]; then
+    echo "true"
+  else
+    echo "false"
+  fi
+}
+
+local_files_subcommand () {
+  if [ "$(supports_local_file_option)" == "true" ] && [ "$FETCH_REMOTE_RESOURCES" == "false" ]; then
+    echo "--local-files /root"
+  else
+    echo ""
+  fi
+}
+
+tailoring_subcommand () {
+  if [ -n "$POLICY_TAILORING_PATH" ]; then
+    echo "--tailoring-file $POLICY_TAILORING_PATH"
+  else
+    echo ""
+  fi
+}
+
+results_path () {
+  echo "$TMP_DIR/results.xml"
+}
+
+results_bzip_path () {
+  echo "$(results_path).bz2"
+}
+
+bzip_command () {
+  echo "/usr/bin/env bzip2 $(results_path)"
+}
+
+scan_command () {
+  local profile=""
+  if [ -n "$POLICY_PROFILE" ]; then
+    profile="--profile $POLICY_PROFILE"
+  fi
+  local fetch_remote_resources=""
+  if [ "$FETCH_REMOTE_RESOURCES" == "true" ]; then
+    fetch_remote_resources="--fetch-remote-resources"
+  fi
+  echo "oscap xccdf eval $fetch_remote_resources $(local_files_subcommand) $profile $(tailoring_subcommand) --results-arf $(results_path) $POLICY_CONTENT_PATH"
+}
+
+run_scan () {
+  local env_vars=$(scan_command_env_vars)
+  local command=$(scan_command)
+  local output=""
+  local status=0
+  if [ -n "$env_vars" ]; then
+    output=$(env $env_vars bash -c "$command" 2>&1)
```

Comment:
Do we need the intermediate shell?

Reply:
This is the only way which worked for me to run a subcommand and set the env variables which won't affect anything but the subcommand.